### PR TITLE
Add tutorial for process_nwb

### PR DESCRIPTION
**Pull request by hlillemark**
Changed tutorial python file to be formatted compatible with sphinx gallery, and added sphinx gallery configurations.

### CODE REVIEW

**Comment by hlillemark** [comment]
Not sure why the test is failing saying that it could not import matplotlib

File: examples/plot_ECoG_preprocessing_tutorial.py [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+# - The wavelet transform outputs the analytical amplitude, which is complex valued.
+# - I'm plotting the magnitude of analytical amplitude which is a proxy of band power
+
+from process_nwb import wavelet_transform
+
+# %% 
+#
+
+
+tf_data, _, ctr_freq, bw = wavelet_transform.wavelet_transform(nth_data, new_sample_rate, filters='rat', hg_only=True, X_fft_h=None, npad=None)
```
Comment (JesseLivezey) [inline examples/plot_ECoG_preprocessing_tutorial.py:169]:
```suggestion
tf_data, _, ctr_freq, bw = wavelet_transform.wavelet_transform(nth_data, new_sample_rate, filters='rat', hg_only=True)
```

File: examples/plot_ECoG_preprocessing_tutorial.py [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+
+
+# %%
+#
+
+ctr_freq[:]
```
Comment (JesseLivezey) [inline examples/plot_ECoG_preprocessing_tutorial.py:175]: Remove?

File: examples/plot_ECoG_preprocessing_tutorial.py [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+
+# %%
+#
+
+tf_data.shape
```
Comment (JesseLivezey) [inline examples/plot_ECoG_preprocessing_tutorial.py:180]: Remove?

File: examples/plot_ECoG_preprocessing_tutorial.py [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+# - I'm plotting the magnitude of analytical amplitude which is a proxy of band power
+
+from process_nwb import wavelet_transform
+
+# %% 
+#
+
+
+tf_data, _, ctr_freq, bw = wavelet_transform.wavelet_transform(nth_data, new_sample_rate, filters='rat', hg_only=True, X_fft_h=None, npad=None)
+
```
Comment (JesseLivezey) [inline examples/plot_ECoG_preprocessing_tutorial.py:170]:
```suggestion
tf_data = abs(tf_data)
```

I would just take the ABS here. You should be zscoring the amplitude, not the complex waveform.

File: examples/plot_ECoG_preprocessing_tutorial.py [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+#
+
+num_tf_signals = len(ctr_freq)
+fig, axs = plt.subplots(num_tf_signals, 1, sharex=True, sharey=True, figsize=(15,10))
+fig.subplots_adjust(hspace=0.4)
+fig.tight_layout
+
+time = np.linspace(0, duration, tf_data.shape[0])
+for idx in range(num_tf_signals):
+    sig = np.abs(tf_data[:,0,idx])
```
Comment (JesseLivezey) [inline examples/plot_ECoG_preprocessing_tutorial.py:192]: Take ABS above.

File: examples/plot_ECoG_preprocessing_tutorial.py [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+
+
+num_tf_signals = len(ctr_freq)
+fig, axs = plt.subplots(num_tf_signals, 1, sharex=True, sharey=True,figsize=(15,10))
+fig.subplots_adjust(hspace=0.4)
+fig.tight_layout
+
+time = np.linspace(0, duration, tf_data.shape[0])
+for idx in range(num_tf_signals):
+    sig = np.abs(tf_norm_data[:,0,idx])
```
Comment (JesseLivezey) [inline examples/plot_ECoG_preprocessing_tutorial.py:234]: Take ABS above.

File: examples/plot_ECoG_preprocessing_tutorial.py [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+    if std is None:
+        std = np.std(signal)
+    return (signal - mean)/std
+
+
+# %%
+#
+
+
+tf_norm_data = np.empty(tf_data.shape, dtype='complex')
```
Comment (JesseLivezey) [inline examples/plot_ECoG_preprocessing_tutorial.py:216]: Doesn't need to be complex if you take ABS earlier.

File: examples/plot_ECoG_preprocessing_tutorial.py [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+
+tf_data.shape
+
+# %%
+#
+
+num_tf_signals = len(ctr_freq)
+fig, axs = plt.subplots(num_tf_signals, 1, sharex=True, sharey=True, figsize=(15,10))
+fig.subplots_adjust(hspace=0.4)
+fig.tight_layout
```
Comment (JesseLivezey) [inline examples/plot_ECoG_preprocessing_tutorial.py:188]:
```suggestion
fig.tight_layout()
```

File: examples/plot_ECoG_preprocessing_tutorial.py [hunk 9]
```diff
@@ -0,0 +1,242 @@
...
+# %% 
+#
+
+
+window = kaiser(1024, 2)
+freq, car_pwr = welch(car_data[:,0], fs=new_sample_rate, window=window)
+_, nth_pwr = welch(nth_data[:,0], fs=new_sample_rate, window=window)
+
+fig, axs = plt.subplots(1,2, figsize=(10,4), sharey=True, sharex=True)
+#fig.tight_layout()
```
Comment (JesseLivezey) [inline examples/plot_ECoG_preprocessing_tutorial.py:141]: Remove?

File: examples/plot_ECoG_preprocessing_tutorial.py [hunk 10]
```diff
@@ -0,0 +1,242 @@
...
+
+
+# %%
+#
+
+
+num_tf_signals = len(ctr_freq)
+fig, axs = plt.subplots(num_tf_signals, 1, sharex=True, sharey=True,figsize=(15,10))
+fig.subplots_adjust(hspace=0.4)
+fig.tight_layout
```
Comment (JesseLivezey) [inline examples/plot_ECoG_preprocessing_tutorial.py:230]:
```suggestion
fig.tight_layout()
```

File: examples/plot_ECoG_preprocessing_tutorial.py [hunk 11]
```diff
@@ -0,0 +1,242 @@
...
+- Time-frequency power calculation
+- Normalizing power for each frequency
+
+"""
+
+# %%
+# Imports
+# ------------------------
+
+import process_nwb as ecog
```
Comment (JesseLivezey) [inline examples/plot_ECoG_preprocessing_tutorial.py:21]: I'd just `import process_nwb` rather than aliasing to `ecog`.

**Review by JesseLivezey — CHANGES_REQUESTED** [review]
The tutorial looks pretty good. There are a few formatting problems and then some problems with git.

I made some suggestions for formatting and clarity in the tutorial file.

You've added a bunch of `.DS_store` files which you'll need to remove. `git add *` can sometimes create a mess. I usually suggest just adding the specific files. You should just have the tutorial .py file in this PR.

It also looks like you probably merged master into your changes. That'll cause my changes to appear multiple times which can sometimes lead to problems. It's a better practice to update your master branch locally, then `git rebase -i master` from your PR branch. This will pull in my changes and apply yours on top.

**Comment by hlillemark** [comment]
Thanks for the review, @JesseLivezey, I'll fix all the syntax and stuff and push a new commit with those changes soon. I was wondering about why it doesn't pass the test, however. I reverted my conf.py file so that it matches the one you put in master, and it has an error with not being able to import matplotlib (and I would assume every other library) in the venv that is created when building the tutorial docs, is there a way to fix this?

**Comment by JesseLivezey** [comment]
@hlillemark I think you'll need to add `matplotlib` in 2 places.

First one is at the end of this `conda install ...` line
https://github.com/BouchardLab/process_nwb/blob/0ce40a9c9d98222038605dcd9e38580dc0b3c20b/.travis.yml#L17

and the second one is on a new line (you can put it after `scipy`) here
https://github.com/BouchardLab/process_nwb/blob/0ce40a9c9d98222038605dcd9e38580dc0b3c20b/environment-dev.yml#L3

File: examples/plot_ECoG_preprocessing_tutorial.py (hunk 4 above)
Comment (hlillemark) [reply examples/plot_ECoG_preprocessing_tutorial.py:170]: This works as intended for the first graph

Comment (hlillemark) [reply examples/plot_ECoG_preprocessing_tutorial.py:170]: After applying abs first before further processing

File: examples/plot_ECoG_preprocessing_tutorial.py (hunk 5 above)
Comment (hlillemark) [reply examples/plot_ECoG_preprocessing_tutorial.py:192]: As in works properly here

File: examples/plot_ECoG_preprocessing_tutorial.py (hunk 6 above)
Comment (hlillemark) [reply examples/plot_ECoG_preprocessing_tutorial.py:234]: Abs_First
Original
Adjusts the Y axis after taking abs first on this graph, might not matter, but I'm not sure

Comment (hlillemark) [reply examples/plot_ECoG_preprocessing_tutorial.py:234]: Top graph is the one produced after taking abs first, bottom graph is original

**Comment by hlillemark** [comment]
All the commits are from rebasing, not sure why it did that, but the actual changes are as intended

File: examples/plot_ECoG_preprocessing_tutorial.py (hunk 6 above)
Comment (jthermiz) [reply examples/plot_ECoG_preprocessing_tutorial.py]: @hlillemark Jesse is right. We should take ABS above and zscore amplitude, not complex values. Top graph is correct.

**Comment by hlillemark** [comment]
Fixed everything with git, formatting, testing, and the tutorial itself. Should be good after a final review.

**Pull request merged**